### PR TITLE
Remove AIP-44 from models/variable

### PR DESCRIPTION
**Pull request by rawwar**
related to https://github.com/apache/airflow/issues/44436

 

- [x] ./airflow/models/variable.py: _set
- [x]  ./airflow/models/variable.py: _update
- [x]  ./airflow/models/variable.py: _delete

### CODE REVIEW

File: airflow/models/variable.py [hunk 1]
```diff
@@ -182,7 +181,6 @@ def set(
 
     @staticmethod
     @provide_session
-    @internal_api_call
```
Comment (jscheffl) [inline airflow/models/variable.py:185]: I assume the change you make is the "simplest way" to remove it. Nevertheless one main problem of AIP-44 was that the core needed to be re-structured and is now a bit cluttered. Example here as well: The original `set()` method is just calling the internal api method `_set()`. That was just introduced to keep interface compatibility but have the DB access separation layer.

Can you check how it was implemented prior AIP-44 and rather revert it to this state back? Means that no `_set()` but the `set()` directly carries-out the call and makes the DB execution.

Comment (jscheffl) [reply airflow/models/variable.py:185]: Ah, just barely remembered, it was even me who implemented AIP-44 in this class in #41414

Comment (potiuk) [reply airflow/models/variable.py:185]: Yeah. We should merge those methods.

Comment (potiuk) [reply airflow/models/variable.py:185]: We can merge it as follow up :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**